core/storage/user/gae_models.py

```diff
@@ -220,7 +222,7 @@ def get_new_id(cls, unused_entity_name):
                 of attempts.
         """
         for _ in python_utils.RANGE(base_models.MAX_RETRIES):
-            new_id = ''.join(
+            new_id = 'uid_%s' % ''.join(
                 random.choice(string.ascii_lowercase)
                 for _ in python_utils.RANGE(USER_ID_LENGTH))
```

Comment:
Should this be USER_ID_RANDOM_PART_LENGTH?

You'll probably also want a backend unit test to ensure that when a new user ID is generated, its length matches USER_ID_LENGTH.

Reply:
Done.